### PR TITLE
9480 RADx MARS de-identification

### PR DESCRIPTION
**Pull request by victor-chaparro**
This PR adds a RADx MARS receiver transform schema that de-identifies all PII data. The list of fields we de-identify can be found on NIH's guide to de-identify HL7 v2 messages [here](https://www.nibib.nih.gov/covid-19/radx-tech-program/mars/hl7-message-de-identification-for-sending-to-hhs-protect).

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. run `./gradlew reloadSettings`
3. Send a test report using the attached Postman collection
[RADx MARS DI.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11627521/RADx.MARS.DI.postman_collection.txt)
4. Verify that the request was accepted
5. Open the generated HL7 file in the local sftp folder.
6. The report sent should have generated two messages one with PII data and the other with de-identified data
7. Verify the fields in the list [here](https://www.nibib.nih.gov/covid-19/radx-tech-program/mars/hl7-message-de-identification-for-sending-to-hhs-protect) have been de-identified.
8. Optional: Compare the two messages to check which fields have been removed and de-identified

## Changes
- Added RADx MARS schema to de-identify PII data
- Updated integration tests

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #9480 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Review by jack-h-wang — COMMENTED** [review]
The integration tests appear to throw eight warnings around formatting a phone number. I'm not sure if this needs to be addressed or if it's OK as is since it doesn't seem to cause an outright failure?

**Comment by victor-chaparro** [comment]
@jack-h-wang I'm setting the values to empty for now. While we investigate if there's a way to suppress those exceptions.

**Review by jack-h-wang — APPROVED** [review]
As far as I can tell, the fields that should be redacted/removed have been. I ran into an exception in the send process, but I think that's covered by #9576 and not directly an issue with the work here.

File: prime-router/metadata/hl7_mapping/common/datatype/xtn-telecom.yml [hunk 1]
```diff
@@ -52,11 +52,23 @@ elements:
     value: [ '%resource.value.getPhoneNumberAreaCode()' ]
     hl7Spec: [ '%{hl7TelecomPath}-6' ]
 
+  # RADx MARS format wnats this value to be <deidentified>, but Hapi library does not allow that so setting it to empty
```
Comment (arnejduranovic) [inline prime-router/metadata/hl7_mapping/common/datatype/xtn-telecom.yml:55]: wants*

File: prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base-deidentified.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+hl7Class: ca.uhn.hl7v2.model.v251.message.ORU_R01
+extends: RADx_MARS-base
+elements:
+  # PID 3.1
+  - name: patient-identifier-list
+    constants:
+      hl7CXField: '%{hl7PIDField}-3'
+      deidentifiedValue: '"<deidentified>"'
```
[the remark this answers is not in the document]
Comment (victor-chaparro) [reply prime-router/metadata/hl7_mapping/ORU_R01/RADx_MARS/RADx_MARS-base-deidentified.yml:8]: I tried doing that, but if I do that, name, address, and telecom will get overridden everywhere and we only want patient data to say <deidentified>.

File: prime-router/metadata/hl7_mapping/common/datatype/xtn-telecom.yml (hunk 1 above)
Comment (victor-chaparro) [reply prime-router/metadata/hl7_mapping/common/datatype/xtn-telecom.yml]: I'm still not sure what needs to be done to allow that. There's a separate ticket to investigate that. Should I add TODO and reference that ticket?

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**